src/main/java/ProjectOcean/Model/CoursePlanningSystem.java

```diff
@@ -1,156 +1,134 @@
     }
 
     /**
-     * @param id is a UUID for a specific course
+     * @param course is a UUID for a specific course
      * @return returns the StudyPoints for the specified UUID
      */
-    public String getStudyPoints(UUID id){
-        return courses.get(id).getStudyPoints();
+    public String getStudyPoints(ICourse course){
+        return course.getStudyPoints();
     }
 
     /**
-     * @param id is a UUID for a specific course
+     * @param course is a UUID for a specific course
      * @return returns the StudyPeriod for the specified UUID
      */
-    public String getStudyPeriod(UUID id){
-        return courses.get(id).getStudyPeriod();
+    public String getStudyPeriod(ICourse course){
+        return course.getStudyPeriod();
     }
 
     /**
-     * @param id is a UUID for a specific course
+     * @param course is a UUID for a specific course
      * @return returns the Examinaot for the specified UUID
      */
-    public String getExaminator(UUID id){
-        return courses.get(id).getExaminator();
+    public String getExaminator(ICourse course){
+        return course.getExaminator();
     }
 
     /**
-     * @param id is a UUID for a specific course
+     * @param course is a UUID for a specific course
      * @return returns the ExaminationMeans for the specified UUID
      */
-    public String getExaminationMeans(UUID id){
-        return courses.get(id).getExaminationMeans();
+    public String getExaminationMeans(ICourse course){
+        return course.getExaminationMeans();
     }
 
     /**
-     * @param id is a UUID for a specific course
+     * @param course is a UUID for a specific course
      * @return returns the Language for the specified UUID
      */
-    public String getLanguage(UUID id){
-        return courses.get(id).getLanguage();
+    public String getLanguage(ICourse course){
+        return course.getLanguage();
     }
 
     /**
-     * @param id is a UUID for a specific course
+     * @param course is a UUID for a specific course
      * @return returns a list of required courses for a specific course defined by a UUID
      */
-    public List<UUID> getRequiredCourses(UUID id){
-        Iterator<Course> iterator = courses.get(id).getRequiredCourses().iterator();
-        List<UUID> uuids = new ArrayList<>();
-
-        while(iterator.hasNext()) {
-            uuids.add(iterator.next().getId());
-        }
-        return uuids;
+    public List<String> getRequiredCourses(ICourse course){
+        return Collections.unmodifiableList(course.getRequiredCourses());
     }
 
     /**
-     * @param id is a UUID for a specific course
+     * @param course is a UUID for a specific course
      * @return returns the CoursePMLink for the specified UUID
      */
-    public String getCoursePMLink(UUID id){
-        return courses.get(id).getCoursePMLink();
+    public String getCoursePMLink(ICourse course){
+        return course.getCoursePMLink();
     }
 
```

Comment:
Most, if not all of these are unessasary with a little refactoring, all methods exist within ICourse itself. Change all usages of them.